riscv/src/register/mstatus.rs

```diff
@@ -84,30 +93,75 @@ impl Mstatus {
         self.bits & (1 << 1) != 0
     }
 
+    /// Update Supervisor Interrupt Enable
+    ///
+    /// Note this updates the [`Mstatus`] value, but does not affect the mstatus
+    /// CSR itself. See [`set_sie`]/[`clear_sie`] to directly update the CSR.
+    #[inline]
+    pub fn update_sie(&self, sie: bool) -> Self {
+        self.bf_insert(1, 1, sie as usize)
```

Comment:
perhaps using a `bit_set(pos: usize, val:bool`? it would be slightly faster when width is 1

Reply:
I'd be surprised if it made any difference - `bf_insert` is inlined so the compiler can see the constants and optimize accordingly.